occurrences/tests/test_event_api_services.py

```diff
@@ -15,15 +16,34 @@ def test_send_event_republish(
     mock_update_event_data,
     p_event,
 ):
-    OccurrenceFactory(p_event=p_event)
+    OccurrenceFactory.create_batch(3, p_event=p_event, cancelled=False)
+
+    first_occurrence = p_event.occurrences.order_by("start_time").first()
+    last_occurrence = p_event.occurrences.order_by("end_time").last()
+    event_start = first_occurrence.start_time
+    event_end = last_occurrence.end_time
+
+    # Set a logic enrolment_start
+    p_event.enrolment_start = first_occurrence.start_time
+    p_event.save()
+
+    enrolling_ends = last_occurrence.start_time - timedelta(
+        days=p_event.enrolment_end_days
+    )
     send_event_republish(p_event)
 
     called_linked_event_id = mock_update_event_to_linkedevents_api.call_args[0][0]
     event_obj = mock_update_event_to_linkedevents_api.call_args[0][1]
     assert called_linked_event_id == p_event.linked_event_id
-    assert event_obj["end_time"] == format_linked_event_datetime(
-        p_event.get_enrolment_end_time_from_occurrences()
+    assert event_obj["start_time"] == format_linked_event_datetime(event_start)
+    assert event_obj["end_time"] == format_linked_event_datetime(event_end)
+    assert event_obj["enrolment_start_time"] == format_linked_event_datetime(
+        p_event.enrolment_start
+    )
+    assert event_obj["enrolment_end_time"] == format_linked_event_datetime(
+        enrolling_ends
     )
+    assert event_obj["enrolment_end_time"] is not None
```

Comment:
This seems like an unnecessary check as `format_linked_event_datetime` does not seem to allow or return null values. Then again, I think you could consider changing it to actually accept null values to get rid of tons of those checks like `x = format_linked_event_datetime(dt) if dt else None` 🙂

Reply:
In that test case I really want to check that it is not None

Reply:
I'll still add a null check to that `format_linked_event_datetime` -function.